### ISSUE TITLE
Using Tensorboard in GluonTS

### ISSUE THREAD

**Issue by codingWhale13** [awslabs/gluon-ts#2320]
[Tensorboard](https://www.tensorflow.org/tensorboard) is a tool for keeping track of and visualizing metrics during training. There's also [tensorboardx](https://github.com/lanpa/tensorboardX) which is platform agnostic.
Tensorboard(x) is generally useful and might also come in handy for GluonTS users. 

Logging with Tensorboard is pretty straightforward:

With **MXNet** models, we can add a callback like so:
```
class CustomCallback(Callback):
    def __init__(self, summary_writer):
        super().__init__()
        self.summary_writer = summary_writer

    def on_train_epoch_end(
            self,
            epoch_no: int,
            epoch_loss: float,
            training_network: nn.HybridBlock,
            trainer: gluon.Trainer,
    ) -> bool:
        self.summary_writer.add_scalar("loss_history", epoch_loss, epoch_no)
        return True


writer = GlobalSummaryWriter()  # use tensorboardX.GlobalSummaryWriter
loss_history = CustomCallback(writer)  # callback, which logs the training loss for each epoch
trainer = Trainer(epochs=10, callbacks=[loss_history])

dataset = get_dataset("m4_hourly")
estimator = DeepAREstimator(
    freq=dataset.metadata.freq,
    prediction_length=dataset.metadata.prediction_length,
    trainer=trainer
)
writer.close()
```

**PyTorch Lightning** makes things very easy:
```
class SomeLightningNetwork(SomeTorchModel, pl.LightningModule):
    def training_step(self, batch, batch_idx):
        loss = 0  # calculate loss
        self.log("training_loss", loss)  # log data
        return loss

logger = TensorBoardLogger(save_dir="some_path")  # use pytorch_lightning.loggers.TensorBoardLogger
trainer = pl.Trainer(logger=logger)

model = SomeLightningNetwork()
trainer.fit(model)
```

Maybe logging with Tensorboard could be supported directly in the `Estimator` class, independent of the ML framework used.
Then again, it already is pretty easy to use as was shown above. So, mentioning how to use Tensorboard in a tutorial might suffice. For example, the approach in PyTorch Lightning could be added to [this](https://ts.gluon.ai/stable/tutorials/advanced_topics/howto_pytorch_lightning.html#) tutorial.

**Comment by lostella** [comment]
> Maybe logging with Tensorboard could be supported directly in the Estimator class, independent of the ML framework used.

I think I would avoid enforcing that from the start.

For MXNet models, would it make sense to have a `TensorBoardLogger` callback that does all of what you have in the snippet? Essentially, construct its own `GlobalSummaryWriter` (and also have an `on_validation_epoch_end` hook)

**Comment by codingWhale13** [comment]
Yes, we could do that. I guess it would be convenient for a few users if such a callback were already provided. It would be very similar to the `TrainingHistory` callback (see [here](https://github.com/awslabs/gluonts/blob/a8259505b572ebdb32015569b939b476a0cc215b/src/gluonts/mx/trainer/callback.py#L307-L331)).

But: This would introduce a new dependency for tensorboardX which probably makes most sense in `requirements/requirements-mxnet.txt`. If that's fine, I can open a PR, adding the `TensorBoardLogger` you described.

**Comment by jaheba** [comment]
Good question where we would want to put it, the problem being the optional dependency. 

But let's not add tensorboardx as a mxnet dependency, that file should probably deleted anyhow.
